IntelliJ plugin not compatible with IntelliJ IDEA 2020.1

**Description:**
I've updated the IntelliJ IDEA (Community Version) to 2020.1 version and the Ballerina IntelliJ plugin 1.2.1 is not compatible with the IDEA 2020.1

I get the following error:
```
com.intellij.diagnostic.PluginException: Cannot create class io.ballerina.plugins.idea.runconfig.application.BallerinaApplicationRunConfigurationType [Plugin: io.ballerina]
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:637)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.instantiateClass(ExtensionComponentAdapter.java:54)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter$SimpleConstructorInjectionAdapter.instantiateClass(XmlExtensionAdapter.java:132)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.createInstance(ExtensionComponentAdapter.java:45)
	at com.intellij.openapi.extensions.impl.XmlExtensionAdapter.createInstance(XmlExtensionAdapter.java:68)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapter(ExtensionPointImpl.java:446)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:401)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensionList(ExtensionPointImpl.java:232)
	at com.intellij.openapi.extensions.ExtensionPointName.getExtensionList(ExtensionPointName.java:42)
	at com.intellij.execution.impl.RunConfigurationSchemeManager$converters$2.invoke(RunConfigurationSchemeManager.kt:26)
	at com.intellij.execution.impl.RunConfigurationSchemeManager$converters$2.invoke(RunConfigurationSchemeManager.kt:22)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.intellij.execution.impl.RunConfigurationSchemeManager.getConverters(RunConfigurationSchemeManager.kt)
	at com.intellij.execution.impl.RunConfigurationSchemeManager.readData(RunConfigurationSchemeManager.kt:60)
	at com.intellij.execution.impl.RunConfigurationSchemeManager.createScheme(RunConfigurationSchemeManager.kt:48)
	at com.intellij.execution.impl.RunConfigurationSchemeManager.createScheme(RunConfigurationSchemeManager.kt:22)
	at com.intellij.configurationStore.LazySchemeProcessor.createScheme$default(scheme-impl.kt:65)
	at com.intellij.configurationStore.schemeManager.SchemeLoader.loadScheme(schemeLoader.kt:175)
	at com.intellij.configurationStore.schemeManager.SchemeManagerImpl$loadSchemes$isLoadOnlyFromProvider$2.invoke(SchemeManagerImpl.kt:194)
	at com.intellij.configurationStore.schemeManager.SchemeManagerImpl$loadSchemes$isLoadOnlyFromProvider$2.invoke(SchemeManagerImpl.kt:43)
	at com.intellij.configurationStore.SchemeManagerIprProvider.processChildren(SchemeManagerIprProvider.kt:48)
	at com.intellij.configurationStore.schemeManager.SchemeManagerImpl.loadSchemes(SchemeManagerImpl.kt:192)
	at com.intellij.configurationStore.schemeManager.SchemeManagerImpl.reload(SchemeManagerImpl.kt:253)
	at com.intellij.execution.impl.RunManagerImpl.loadState(RunManagerImpl.kt:748)
	at com.intellij.execution.impl.RunManagerImpl.loadState(RunManagerImpl.kt:72)
	at com.intellij.configurationStore.ComponentStoreImpl.doInitComponent(ComponentStoreImpl.kt:439)
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:373)
	at com.intellij.configurationStore.ComponentStoreImpl.initPersistenceStateComponent(ComponentStoreImpl.kt:138)
	at com.intellij.configurationStore.ComponentStoreImpl.initComponent(ComponentStoreImpl.kt:103)
	at com.intellij.configurationStore.ComponentStoreWithExtraComponents.initComponent(ComponentStoreWithExtraComponents.kt:50)
	at com.intellij.serviceContainer.ComponentManagerImpl.initializeComponent$intellij_platform_serviceContainer(ComponentManagerImpl.kt:343)
	at com.intellij.serviceContainer.ServiceComponentAdapter.createAndInitialize(ServiceComponentAdapter.kt:58)
	at com.intellij.serviceContainer.ServiceComponentAdapter.access$createAndInitialize(ServiceComponentAdapter.kt:15)
	at com.intellij.serviceContainer.ServiceComponentAdapter$doCreateInstance$$inlined$use$lambda$1.run(ServiceComponentAdapter.kt:47)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:625)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:570)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:186)
	at com.intellij.serviceContainer.ServiceComponentAdapter.doCreateInstance(ServiceComponentAdapter.kt:46)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstanceUncached(BaseComponentAdapter.kt:115)
	at com.intellij.serviceContainer.BaseComponentAdapter.getInstance(BaseComponentAdapter.kt:69)
	at com.intellij.serviceContainer.ComponentManagerImpl.doGetService(ComponentManagerImpl.kt:396)
	at com.intellij.serviceContainer.ComponentManagerImpl.getService(ComponentManagerImpl.kt:379)
	at com.intellij.execution.impl.ProjectRunConfigurationInitializer.serviceCreated(ProjectRunConfigurationInitializer.kt:28)
	at com.intellij.openapi.project.ProjectLoadHelper$Companion$notifyThatComponentCreated$$inlined$runHandler$1.accept(projectLoader.kt:110)
	at com.intellij.openapi.project.ProjectLoadHelper$Companion$notifyThatComponentCreated$$inlined$runHandler$1.accept(projectLoader.kt)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processWithPluginDescriptor(ExtensionPointImpl.java:285)
	at com.intellij.openapi.project.ProjectLoadHelper$Companion.notifyThatComponentCreated(projectLoader.kt:114)
	at com.intellij.openapi.project.ProjectLoadHelper.notifyThatComponentCreated(projectLoader.kt)
	at com.intellij.openapi.project.impl.ProjectImpl.init(ProjectImpl.java:281)
	at com.intellij.openapi.project.impl.ProjectManagerImpl.initProject(ProjectManagerImpl.java:320)
	at com.intellij.platform.PlatformProjectOpenProcessorKt.convertAndLoadProject(PlatformProjectOpenProcessor.kt:413)
	at com.intellij.platform.PlatformProjectOpenProcessorKt.prepareProject(PlatformProjectOpenProcessor.kt:308)
	at com.intellij.platform.PlatformProjectOpenProcessorKt.access$prepareProject(PlatformProjectOpenProcessor.kt:1)
	at com.intellij.platform.PlatformProjectOpenProcessor$Companion$openExistingProject$$inlined$runInAutoSaveDisabledMode$lambda$1.invoke(PlatformProjectOpenProcessor.kt:181)
	at com.intellij.platform.PlatformProjectOpenProcessor$Companion$openExistingProject$$inlined$runInAutoSaveDisabledMode$lambda$1.invoke(PlatformProjectOpenProcessor.kt:58)
	at com.intellij.platform.ProjectUiFrameAllocator$run$1$$special$$inlined$createModalTask$1.run(progress.kt:37)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:930)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:480)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:77)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$null$3(ProgressRunner.java:233)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:164)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:625)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:570)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:61)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:151)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:233)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:595)
	... 70 more
Caused by: java.lang.NoSuchFieldError: HideRight
	at icons.BallerinaIcons.<clinit>(BallerinaIcons.java:47)
	at io.ballerina.plugins.idea.runconfig.application.BallerinaApplicationRunConfigurationType.<init>(BallerinaApplicationRunConfigurationType.java:34)
	... 75 more
```

With this error, we cannot open any ballerina file. I disabled the plugin and then able to open `.bal` files.

**Affected Versions:**
IntelliJ IDEA 2020.1
Ballerina IDEA plugin 1.2.1

--- a/tool-plugins/intellij/gradle.properties
+++ b/tool-plugins/intellij/gradle.properties
@@ -15,4 +15,4 @@
 #
 ballerinaPluginName=ballerina-intellij-idea-plugin
 ballerinaPluginVersion=1.2.1
-ideaVersion=2019.2
+ideaVersion=2020.1

--- a/tool-plugins/intellij/gradle.properties
+++ b/tool-plugins/intellij/gradle.properties
@@ -14,5 +14,5 @@
 #  limitations under the License.
 #
 ballerinaPluginName=ballerina-intellij-idea-plugin
-ballerinaPluginVersion=1.2.1
+ballerinaPluginVersion=1.2.2
 ideaVersion=2020.1

--- a/tool-plugins/intellij/src/main/java/icons/BallerinaIcons.java
+++ b/tool-plugins/intellij/src/main/java/icons/BallerinaIcons.java
@@ -34,7 +34,7 @@ public class BallerinaIcons {
     public static final Icon ICON = FILE;
     public static final Icon APPLICATION_RUN = createIconWithShift(ICON, AllIcons.Nodes.RunnableMark);
     public static final Icon RUN = AllIcons.RunConfigurations.TestState.Run;
-    public static final Icon PACKAGE = PlatformIcons.DIRECTORY_CLOSED_ICON;
+    public static final Icon PACKAGE = PlatformIcons.FOLDER_ICON;
 
     public static final Icon TEST = AllIcons.RunConfigurations.TestState.Green2;
     public static final Icon RECURSIVE = AllIcons.Gutter.RecursiveMethod;
@@ -44,15 +44,11 @@ public class BallerinaIcons {
     public static final Icon PARAMETER = AllIcons.Nodes.Parameter;
     public static final Icon NAMESPACE = AllIcons.Json.Array;
     public static final Icon SERVICE = AllIcons.Nodes.Static;
-    public static final Icon RESOURCE = AllIcons.General.HideRight;
     public static final Icon ANNOTATION = AllIcons.Nodes.Annotationtype;
     public static final Icon WORKER = AllIcons.Nodes.Rw_access;
     public static final Icon ENDPOINT = AllIcons.Ide.UpDown;
     public static final Icon GLOBAL_ENDPOINT = new LayeredIcon(ENDPOINT, AllIcons.Nodes.StaticMark);
     public static final Icon TYPE = AllIcons.Nodes.Artifact;
-
-    public static final Icon PUBLIC_FIELD = createPublicFieldIcon(AllIcons.Nodes.Advice);
-    public static final Icon PRIVATE_FIELD = createPrivateFieldIcon(AllIcons.Nodes.Advice);
 
     /**
      * Contains icons used for the ballerina diagram editor(referred by plugin.xml).

--- a/tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/preloading/LSPUtils.java
+++ b/tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/preloading/LSPUtils.java
@@ -63,7 +63,7 @@ import static io.ballerina.plugins.idea.BallerinaConstants.SYS_PROP_LS_TRACE;
  */
 public class LSPUtils {
 
-    private static BallerinaAutoDetectNotifier autoDetectNotifier = new BallerinaAutoDetectNotifier();
+    private static final BallerinaAutoDetectNotifier autoDetectNotifier = new BallerinaAutoDetectNotifier();
     private static final Logger LOG = Logger.getInstance(LSPUtils.class);
 
     /**

--- a/tool-plugins/intellij/src/main/resources/META-INF/plugin.xml
+++ b/tool-plugins/intellij/src/main/resources/META-INF/plugin.xml
@@ -1,7 +1,7 @@
 <idea-plugin>
     <id>io.ballerina</id>
     <name>Ballerina</name>
-    <version>1.2.1</version>
+    <version>1.2.2</version>
     <vendor email="ballerinalang@wso2.com" url="https://ballerina.io/">BallerinaLang</vendor>
 
     <description><![CDATA[
@@ -13,9 +13,7 @@
 
     <change-notes><![CDATA[
       <ul>
-        <li> Fixed invalid cursor placement after ballerina strings being hard wrapped while typing. </li>
-        <li> Fixed potential NPE when working with ballerina documentation in some contexts. </li>
-        <li> Improved error notifications on language server startup failures. </li>
+        <li> Fixed compatibility issues with IntelliJ IDEA 2020.1 platform. </li>
       </ul>
     ]]>
     </change-notes>

